draft-pauly-quic-datagram.md

```diff
@@ -185,11 +185,13 @@ ACK frames (within the limits specified by max_ack_delay) in reponse to receivin
 packets that only contain DATAGRAM frames, since the timing of these
 acknowledgements is not used for loss recovery.
 
-If a sender detects that a packet containing a specific DATAGRAM frame has been
-lost, the implementation MAY notify the application that the datagram was lost.
-Similarly, if a packet containing a DATAGRAM frame is acknowledged, the
-implementation MAY notify the application that the datagram was successfully
-transmitted and received.
+If a sender detects that a packet containing a specific DATAGRAM frame might
+have been lost, the implementation MAY notify the application that it believes
```

Comment:
You use `MAY` here, but I feel `SHOULD` might be better. The transport has this extra info and I see this as one of the primary reasons for using QUIC instead of raw UDP (besides encryption). It'd be a shame for it not to indicate this information up.

Reply:
I don't think we SHOULD anything wrt. API design. But MAY also sounds permissive so I'd rather say "can".

Reply:
I agree that we shouldn't have a SHOULD on API design. MAY seems like the right choice though: https://tools.ietf.org/html/rfc2119#section-5